### PR TITLE
add flag to control whether padding is displayed or not in logs

### PR DESCRIPTION
**Pull request by Ranapop**

### CODE REVIEW

File: examples/cfq/input_pipeline.py [hunk 1]
```diff
@@ -144,10 +144,12 @@ def get_example_length(self, example: ExampleType) -> tf.Tensor:
...
     return example_len
 
-  def indices_to_sequence_string(self, indices: jnp.ndarray) -> Text:
+  def indices_to_sequence_string(self,
+                                 indices: jnp.ndarray,
+                                 keep_padding: bool = True) -> Text:
     """Transforms a list of vocab indices into a string
         (e.g. from token indices to question/query)"""
-    tokens = [self.i2w[i].decode() for i in indices]
+    tokens = [self.i2w[i].decode() for i in indices if keep_padding or i!=0]
```
Comment (marcvanzee) [inline examples/cfq/input_pipeline.py:153]: I don't really understand how this works, what padding will it remove? Currently, it seems that if `keep_padding == False`, then this will output the entire sequence except the first index. Don't you want to remove padding tokens later in the sequence as well?

File: examples/cfq/input_pipeline.py [hunk 2]
```diff
@@ -144,10 +144,12 @@ def get_example_length(self, example: ExampleType) -> tf.Tensor:
       example_len = tf.math.maximum(question_len, query_len)
     return example_len
 
-  def indices_to_sequence_string(self, indices: jnp.ndarray) -> Text:
+  def indices_to_sequence_string(self,
+                                 indices: jnp.ndarray,
+                                 keep_padding: bool = True) -> Text:
```
Comment (marcvanzee) [inline examples/cfq/input_pipeline.py]: Why not make `keep_padding` False by default? (seems to be what you usually want, and then you don't have to give it as an argument below).

**Review by marcvanzee — CHANGES_REQUESTED** [review]

File: examples/cfq/input_pipeline.py (hunk 1 above)
Comment (Ranapop) [reply examples/cfq/input_pipeline.py:153]: The padding token has index zero. This convention is used throughout the code (also when the padding is done). What this piece of code does is filter out the zero-valued indices, the indices of padding tokens, when keep_padding = False. I've explained this in the function description now.

File: examples/cfq/input_pipeline.py (hunk 2 above)
Comment (Ranapop) [reply examples/cfq/input_pipeline.py]: will change

File: examples/cfq/input_pipeline.py (hunk 1 above)
Comment (marcvanzee) [reply examples/cfq/input_pipeline.py:153]: Ohh right, i is the index. Sorry I was confused. It makes sense now, thanks for explaning!

File: examples/cfq/input_pipeline.py [hunk 3]
```diff
@@ -208,18 +211,16 @@ def get_bucketed_batches(self,
...
-  print('Vocab size')
-  print(data_source.vocab_size)
+
+  # Print queries
+  batch_no = 1
+  for batch in tfds.as_numpy(train_batches):
+    queries = batch[constants.QUERY_KEY]
+    print('Batch no ',batch_no)
+    for query in queries:
+      print(data_source.indices_to_sequence_string(query, keep_padding=False))
```
Comment (marcvanzee) [inline examples/cfq/input_pipeline.py]:
```suggestion
      print(data_source.indices_to_sequence_string(query))
```

**Review by marcvanzee — APPROVED** [review]
